### PR TITLE
Enable local address and leak sanitizer build and fix resource release in tests

### PR DESCRIPTION
**Pull request by EvgenyMekhanik**
Enable local address and leak sanitizer build and fix
resource release in tests.

### CODE REVIEW

File: test/tarantool_disconnect.c [hunk 1]
```diff
@@ -14,7 +14,8 @@ int main() {
 	tnt_set(tnt, TNT_OPT_RECV_BUF, 0); // Disable buffering for recv
 	tnt_connect(tnt);
 
-	while (1) {
+	for (int i = 0; i < 10; i++)
+	{
```
Comment (Totktonada) [inline test/tarantool_disconnect.c:18]: What is the point of the change? AFAIU it is the test is for manual testing: you run the server, run the client and perform several stop / start of the server. The client should reconnect and process responses correctly (see #41).

I reverted this hunk, built the connector and the test with ASAN and after Ctrl+C it does not show anything bad on the terminal.

File: CMakeLists.txt [hunk 2]
```diff
@@ -79,6 +79,18 @@ if (ENABLE_BUNDLED_DOCS)
...
+    if (CMAKE_COMPILER_IS_GNUCC)
+        message(FATAL_ERROR
+            "\n"
+            " Tarantool-c does not support GCC's AddressSanitizer, use clang\n"
+            )
+    endif()
+
+    set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -fsanitize=address -fsanitize=leak")
+endif(ENABLE_ASAN)
+
```
Comment (locker) [inline CMakeLists.txt:93]: Doesn't gcc support address sanitizer nowadays? On my system (gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)), according to the manual, it does.

**Review by locker — CHANGES_REQUESTED** [review]

File: test/tarantool_disconnect.c (hunk 1 above)
Comment (EvgenyMekhanik) [reply test/tarantool_disconnect.c:18]: Ok, i revert this part.

File: CMakeLists.txt (hunk 2 above)
Comment (EvgenyMekhanik) [reply CMakeLists.txt:93]: GCC support address sanitizer nowadays, this part is same as in tarantool. Ok i fixed it

**Review by locker — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]
The changes looks logical, thanks!

**Pull request merged**